tests/unit_tests/blas/batch/gemm_batch_stride.cpp

```diff
@@ -63,10 +63,10 @@ int test(device *dev, oneapi::mkl::layout layout, int64_t batch_size) {
     lda = std::max(m, k);
     ldb = std::max(n, k);
     ldc = std::max(m, n);
-    alpha = rand_scalar<fp>();
-    beta = rand_scalar<fp>();
+    alpha = rand_scalar<Ts>();
+    beta = rand_scalar<Ts>();
 
-    if ((std::is_same<fp, float>::value) || (std::is_same<fp, double>::value)) {
+    if ((std::is_same<Ts, float>::value) || (std::is_same<Ts, double>::value)) {
```

Comment:
I don't think this is doing what you want. If `Ts` is half, for example, you may get conjtrans in line 76. Probably it makes sense to reverse the sense of the if and check for `complex<float>` or `complex<double>` in the if and everything else in the else.

Reply:
I see your point. I have flipped the check and applied it to all the tests